osu.Game.Rulesets.Mania/Beatmaps/Patterns/Legacy/DistanceObjectPatternGenerator.cs

```diff
@@ -511,7 +519,7 @@ private void addToPattern(Pattern pattern, int column, double startTime, double
                     Duration = endTime - startTime,
                     Column = column,
                     Samples = HitObject.Samples,
-                    NodeSamples = (HitObject as IHasRepeats)?.NodeSamples
+                    NodeSamples = nodeSamplesAt(startTime)
```

Comment:
Are you sure about this? The reason I didn't merge this PR myself is that I don't think this is actually the case for converted beatmaps.

Reply:
I didn't check stable code, just a couple of hitsounded beatmaps (that looked to match). May be worth doing if you're unsure.